### PR TITLE
 chore: update to latest changes in tangle 

### PR DESCRIPTION
**Pull request by shekohex**
This pull request includes several updates across multiple files, focusing on dependency updates, code refactoring, and metadata extraction improvements. The most important changes include upgrading dependencies, refactoring hooks and job extraction logic, and introducing new metadata extraction functions.

Mainly, the new changes we are introducing in the https://github.com/tangle-network/tangle/pull/794 removes the notion of separate hooks and instead unifies everything in one filed called `manager` in the Service Blueprint. This field will be set to the EVM Smart contract that manages the lifecycle of a the blueprint instances.

```toml
# Cargo.toml

[package.metadata.blueprint]
manager = { evm = "IncredibleSquaringBlueprint" }
```

No more `#[job(verifier(evm = "IncredibleSquaringBlueprint"))]` on top of each job. This manager has a `onJobResult` and `onJobCall` hooks where you can do your normal validation and verification if you want. (by default they are no_op)

### Dependency Updates:
* [`Cargo.toml`](diffhunk://#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542L60-R60): Updated `tangle-subxt` dependency from version `0.3.0` to `0.4.0`.
* [`blueprint-metadata/Cargo.toml`](diffhunk://#diff-9edc95e9d1a3c1b74c180c9be6aa02dfaddde89d9e571adcdc8200a6b5683747R14): Added `serde` as a workspace dependency.

### Code Refactoring:
* [`blueprint-metadata/src/lib.rs`](diffhunk://#diff-bae1a9955668baf7621f8870316b2ceda7433069c7d5347a21d4e48401444651R8-R11): Refactored imports to include `cargo_metadata` and rearranged existing ones.
* [`blueprint-metadata/src/lib.rs`](diffhunk://#diff-bae1a9955668baf7621f8870316b2ceda7433069c7d5347a21d4e48401444651R277-R309): Replaced `generate_gadget` function with `extract_metadata` and `extract_blueprint_manager` functions for better modularity.
* [`blueprint-metadata/src/lib.rs`](diffhunk://#diff-bae1a9955668baf7621f8870316b2ceda7433069c7d5347a21d4e48401444651L93-L95): Removed `ServiceRegistrationHook` and `ServiceRequestHook` from the `Hook` enum and related logic. [[1]](diffhunk://#diff-bae1a9955668baf7621f8870316b2ceda7433069c7d5347a21d4e48401444651L93-L95) [[2]](diffhunk://#diff-bae1a9955668baf7621f8870316b2ceda7433069c7d5347a21d4e48401444651L185-L216)

### Metadata Extraction:
* [`blueprint-metadata/src/lib.rs`](diffhunk://#diff-bae1a9955668baf7621f8870316b2ceda7433069c7d5347a21d4e48401444651R277-R309): Introduced `extract_metadata` and `extract_blueprint_manager` functions to handle metadata extraction from the package.
* [`blueprints/incredible-squaring/Cargo.toml`](diffhunk://#diff-6eff0bd9ae51c620ed2a0d2b6569a6e3d41dd54a3cbb7f8718a590766632fcf1R54-R56): Added blueprint manager metadata.

### Submodule Updates:
* [`blueprints/incredible-squaring/contracts/lib/forge-std`](diffhunk://#diff-5c8acdecf5973835412bd29409fcabfc7269606eb3bb3104c8a98232e3fec3b8L1-R1): Updated submodule commit reference.
* [`blueprints/incredible-squaring/contracts/lib/tnt-core`](diffhunk://#diff-b880f12a01758f1c5edd8334a27abaa2ef1cc91a7dff4627207e7569a6b18823L1-R1): Updated submodule commit reference.

### CODE REVIEW

**Comment by shekohex** [comment]
Companion PRs
1. https://github.com/tangle-network/blueprint-template/pull/22
2. https://github.com/tangle-network/blueprint-template/pull/23

File: blueprint-metadata/src/lib.rs [hunk 1]
```diff
@@ -324,8 +274,40 @@ fn generate_gadget() -> Gadget<'static> {
...
+}
 
-    let package = find_package(&metadata, &crate_name);
+fn extract_blueprint_manager(package: &Package) -> BlueprintManager {
+    let Some(blueprint) = package.metadata.get("blueprint") else {
+        eprintln!("No blueprint metadata found in the Cargo.toml.");
+        eprintln!("For more information, see:");
+        eprintln!("<TODO>");
+        // For now, we just return an empty blueprint manager, this
+        // will be a hard error in the future.
```
Comment (drewstone) [inline blueprint-metadata/src/lib.rs:291]: Should be a TODO.

**Review by drewstone — APPROVED** [review]
Fix the comment, this looks good.

File: blueprint-metadata/src/lib.rs (hunk 1 above)
Comment (shekohex) [inline blueprint-metadata/src/lib.rs:291]:
```suggestion
        // TODO(@shekohex): make this hard error
```

**Pull request merged**